java/util/src/main/java/com/google/protobuf/util/JsonFormat.java

```diff
@@ -151,7 +158,17 @@ public Printer includingDefaultValueFields() {
      * current {@link Printer}.
      */
     public Printer preservingProtoFieldNames() {
-      return new Printer(registry, includingDefaultValueFields, true);
+      return new Printer(registry, includingDefaultValueFields, true, minified);
+    }
+
+
+    /**
+     * Creates a new {@link Printer} that has a different textGenerator, The json output will be minified.
+     * The new Printer clones all other configurations from the
+     * current {@link Printer}.
+     */
+    public Printer setPrinterTypeUgly(){
```

Comment:
I think we need a better name for this option. It should be consistent with the other options.